services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,43 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (event != NULL) {
+    if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == pdPASS)
+      return ERR_CODE_SUCCESS;
+    else
+      return ERR_CODE_QUEUE_FULL;
+  }
+  else
+    return ERR_CODE_INVALID_ARG;
 
-  return ERR_CODE_SUCCESS;
+  return ERR_CODE_INVALID_QUEUE_MSG;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT_CMD;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t eventBuf;
```

Comment:
in general try to limit the scope of your variables to the smallest scope possible, so move this into the while block